virtual-desktop/src/app/plugin-manager/plugin-factory/iframe/iframe-plugin.component.ts

```diff
@@ -113,18 +138,22 @@ export class IFramePluginComponent {
   }
 
   private addActionsToContextMenu(key: number, source: any, itemsArray: Array<any>){
-    let copy = JSON.parse(JSON.stringify(itemsArray));
-    for(let i = 0; i < copy.length; i++){
-      copy[i].action = () => {
-        source.postMessage({
-          key: key,
-          originCall: 'windowActions.spawnContextMenu',
-          instanceId: this.instanceId,
-          contextMenuItemIndex: i
-        }, '*')
+    try{
+      let copy = JSON.parse(JSON.stringify(itemsArray));
+      for(let i = 0; i < copy.length; i++){
+        copy[i].action = () => {
+          source.postMessage({
+            key: key,
+            originCall: 'windowActions.spawnContextMenu',
+            instanceId: this.instanceId,
+            contextMenuItemIndex: i
+          }, '*')
+        }
       }
+      return copy;
+    }catch(e){
+      return undefined;
```

Comment:
Would it be useful to log the error here too? I'm not familiar with this code base, it's your call

Reply:
Yes, let's put a log here stating what error occurred.